public/index.ts

```diff
@@ -599,7 +599,7 @@ function updateZSliceUI(volume: Volume) {
   const zSlider = document.getElementById("zSlider") as HTMLInputElement;
   const zInput = document.getElementById("zValue") as HTMLInputElement;
 
-  const totalZSlices = volume.z;
+  const totalZSlices = volume.imageInfo.vol_size_z || volume.z;
```

Comment:
This pattern of using `z` as a fallback shows up a bunch when I need the size of the full volume in pixels. (It's required only for z, the special dimension which we currently never downsample.) It's ugly, and indicates that some `Volume` properties might need reshuffling/renaming, and/or that we ought to remove `ImageInfo` as it currently exists from `Volume`.

Reply:
`ImageInfo` in its current form to be isolated to `JsonImageInfoLoader` only and replaced with a more sensible data structure for the modern age of this tool. This may happen in a separate PR to reduce review burden.

Reply:
Tracked by #131